### PR TITLE
Feat/new sign in wizard

### PR DESCRIPTION
**Pull request by Zuhal-Ayob**
#### Work done
- Componentise dashboard
- Create resolver for wizard
- Create wizard component
- Add previous/next buttons
- Benchmarks flag
- Set up routing from registration survey

#### Tests
Does this PR include tests for the changes introduced?
- [x] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: src/app/features/first-login-wizard/first-login-wizard.component.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<div class="govuk-grid-row govuk-!-margin-1 govuk-!-margin-top-4">
+  <div class="asc-summary-box">
+    <div class="govuk-grid-row">
+      <div class="govuk-grid-column-full govuk-util__align-right">
+        <a class="govuk-button govuk-button--link" href="/dashboard">Close</a>
```
Comment (aaron-russell) [inline src/app/features/first-login-wizard/first-login-wizard.component.html:5]: Shouldn't this be a routerlink?
```suggestion
        <a class="govuk-button govuk-button--link" routerLink="/dashboard">Close</a>
```

File: src/app/features/first-login-wizard/first-login-wizard.component.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        >
+        <br />
+        <a
+          *ngIf="!isFirst"
+          href="#"
+          (click)="previousWizard($event)"
+          class="govuk-button--link govuk-!-margin-left-2"
+          >{{ wizards[currentIndex - 1].title }}</a
+        >
+      </div>
```
Comment (aaron-russell) [inline src/app/features/first-login-wizard/first-login-wizard.component.html:41]: Why not put the `*ngIf` round the whole div?
```suggestion
      <div class="govuk-util__inline-block govuk-grid-column-one-half govuk-util__align-left" *ngIf="!isFirst">
        <a
          class="asc-chevron-before asc-no-underline govuk-!-font-size-27 govuk-button--link govuk-!-margin-top-3"
          href="#"
          (click)="previousWizard($event)"
          >Previous</a
        >
        <br />
        <a
          href="#"
          (click)="previousWizard($event)"
          class="govuk-button--link govuk-!-margin-left-2"
          >{{ wizards[currentIndex - 1].title }}</a
        >
      </div>
```

File: src/app/features/first-login-wizard/first-login-wizard.component.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          "
+          href="#"
+          (click)="nextWizard($event)"
+          >Next
+        </a>
+        <br />
+        <a *ngIf="!isLast" href="#" (click)="nextWizard($event)" class="govuk-button--link govuk-!-margin-right-2">{{
+          wizards[currentIndex + 1].title
+        }}</a>
+      </div>
```
Comment (aaron-russell) [inline src/app/features/first-login-wizard/first-login-wizard.component.html:59]: Same here
```suggestion
      <div class="govuk-util__inline-block govuk-grid-column-one-half govuk-util__float-right govuk-util__align-right" *ngIf="!isLast">
        <a
          class="
            asc-chevron-after asc-no-underline
            govuk-button--link
            govuk-!-font-size-27 govuk-!-margin-top-3
            margin-left-6
          "
          href="#"
          (click)="nextWizard($event)"
          >Next
        </a>
        <br />
        <a href="#" (click)="nextWizard($event)" class="govuk-button--link govuk-!-margin-right-2">{{
          wizards[currentIndex + 1].title
        }}</a>
      </div>
```

File: src/app/features/first-login-wizard/first-login-wizard.component.scss [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  padding: 20px 40px;
+
+  // @media only screen and (max-width: 830px) {
+  //   padding: 15px 20px;
+  // }
+
+  // @media only screen and (max-width: 510px) {
+  //   flex-direction: column;
+  //   padding: 30px 20px;
+  // }
```
Comment (aaron-russell) [inline src/app/features/first-login-wizard/first-login-wizard.component.scss:16]: Is this needed if it's commented out?

File: src/app/features/first-login-wizard/first-login-wizard.component.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    event.preventDefault();
+    this.currentIndex += 1;
+    this.updateVariables();
+  }
+
+  public previousWizard(event: Event): void {
+    event.preventDefault();
+    this.currentIndex -= 1;
+    this.updateVariables();
+  }
```
Comment (aaron-russell) [inline src/app/features/first-login-wizard/first-login-wizard.component.ts:41]: Could these not be the same function with the input to + or -?

File: src/app/features/first-login-wizard/first-login-wizard.component.html (hunk 1 above)
Comment (joannafawl) [reply src/app/features/first-login-wizard/first-login-wizard.component.html:5]: Should we always use `routerLink` over `href`? I thought every `<a>` tag needed an `href`

Comment (aaron-russell) [reply src/app/features/first-login-wizard/first-login-wizard.component.html:5]: `routerLink` adds a `href` but wont reload the web app :)

Comment (aaron-russell) [reply src/app/features/first-login-wizard/first-login-wizard.component.html:5]: https://stackoverflow.com/questions/58081360/anchor-href-vs-angular-routerlink for reference

**Review by aaron-russell — APPROVED** [review]

**Pull request merged**